lib/server/services/detector/getOrCreateAllPhysicalDetectorsByName.js

```diff
@@ -14,22 +14,40 @@
 const { getDetectorsByNames } = require('./getDetectorsByNames.js');
 const { createDetector } = require('./createDetector.js');
 const { getDetector } = require('./getDetector.js');
+const DetectorRepository = require('../../../database/repositories/DetectorRepository.js');
+const { Op } = require('sequelize');
 
 /**
- * Find all the detectors corresponding to a list of detectors names. If any of the name do not correspond to a detector, creates it
+ * Find all the physical detectors corresponding to a list of detectors names. If any of the name do not correspond to a detector, creates it
  *
  * @param {string[]} [detectorNames] the list of detectors name
  * @return {Promise<SequelizeDetector[]>} the list of detectors
  */
-exports.getOrCreateAllDetectorsByName = async (detectorNames) => {
+exports.getOrCreateAllPhysicalDetectorsByName = async (detectorNames) => {
```

Comment:
Actually, TST needs to be included here, so you'll need an OR somewhere. IF needed we might name VIRTUAL + PHYSICAL detectors `DataTaking` detector?